catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/NewDataDialogTest.java

```diff
@@ -0,0 +1,220 @@
+
+		onView(withId(R.id.input_edit_text))
+				.perform(typeText(spaceVarName));
+
+		clickOk();
+
+		checkErrorMessageDisplayed(R.string.name_consists_of_spaces_only);
+	}
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testInvalidThenValidVarNameErrorMessage() {
+		String spaceVarName = " ";
+		String validVarName = "name";
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(spaceVarName);
+
+		checkErrorMessageDisplayed(R.string.name_consists_of_spaces_only);
+
+		onView(withId(R.id.input_edit_text))
+				.perform(typeText(validVarName));
+
+		clickOk();
+
+		checkVariableShownInSpinner(validVarName);
+	}
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testValidSpecialCharVarNames() {
+		String underscoreVarName = "var_var";
+		String numberVarName = "var123";
+		String validSpecialCharVarName = "var$";
+		String slashVarName = "var/";
+		String bracketVarName = "var(";
+		String curlyBracketVarName = "var{";
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(underscoreVarName);
+		checkVariableShownInSpinner(underscoreVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(numberVarName);
+		checkVariableShownInSpinner(numberVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(validSpecialCharVarName);
+		checkVariableShownInSpinner(validSpecialCharVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(slashVarName);
+		checkVariableShownInSpinner(slashVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(bracketVarName);
+		checkVariableShownInSpinner(bracketVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(curlyBracketVarName);
+		checkVariableShownInSpinner(curlyBracketVarName);
+	}
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testCancelVarCreation() {
+		clickOnEmptySpinner(variableBrickPosition, R.id.set_variable_spinner);
+
+		clickCancel();
+
+		onBrickAtPosition(variableBrickPosition)
+				.checkShowsText(R.string.brick_variable_spinner_create_new_variable);
+	}
+
+	private void clickOnEmptySpinner(int brickPosition, int spinnerId) {
+		onBrickAtPosition(brickPosition)
+				.onVariableSpinner(spinnerId)
+				.perform(click());
+	}
+
+	private void clickCancel() {
+		onView(withText(R.string.cancel))
+				.perform(click());
+	}
+
+	private void clickOk() {
+		onView(withText(R.string.ok))
+				.perform(click());
+	}
+
+	private void checkErrorMessageDisplayed(int errorMessageId) {
+		onView(withText(errorMessageId))
```

Comment:
since the method name promises more than what it actually does, and its only containing one command, please remove this method